In [2]:
!pip install biopython

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.2/3.2 MB 18.8 MB/s eta 0:00:00


In [4]:
from Bio import SeqIO
from Bio.SeqUtils import gc_fraction

# Parse the file
records = list(SeqIO.parse('sequence.fasta', 'fasta'))
for r in records:
  print(f'ID: {r.id}')
  print(f'Length: {len(r.seq)}')
  print(f'GC: {gc_fraction(r.seq)*100:.2f}%')

# Print first 50 bases of sequence
  print(f'Sequence: {str(r.seq[:50])}...')


ID: AH002844.2
Length: 4969
GC: 63.85%
Sequence: CTCGAGGGGCCTAGACATTGCCCTCCAGAGAGAGCACCCAACACCCTCCA...


# **FASTA File Parser and Sequence Analysis**

#### **Objective**

The objective of this analysis is to download a DNA sequence in FASTA format
from the **NCBI database**, upload it to Google Colab, parse the FASTA file
using **Biopython**, and extract basic sequence information from the DNA sequence, including its sequence ID, length, GC content, and first 50 nucleotides of the sequence

#### **Method**

The **Biopython `SeqIO` module** is used to read and parse the FASTA file.

The FASTA file contains a DNA sequence associated with the **insulin (INS) gene**.

The `gc_fraction()` function from `Bio.SeqUtils` is used to calculate the
GC content of the DNA sequence.

The GC content is calculated using the formula:

$$
GC\% = \frac{G+C}{A+T+G+C}\times100
$$

Only the first 50 nucleotides are displayed as a sequence preview.

#### **Result**

The FASTA sequence was successfully parsed.

**Sequence ID**: AH002844.2

**Sequence Length**: 4969 bp

**GC Content**: 63.85%

**Sequence Preview**: CTCGAGGGGCCTAGACATTGCCCTCCAGAGAGAGCACCCAACACCCTCCA...

#### **Interpretation**

The sequence **AH002844.2** has a length of **4969 base pairs (bp)**.

The calculated **GC content is 63.85%**, which means that approximately
63.85% of the nucleotides in the sequence are **Guanine (G) or Cytosine (C)**.
The remaining approximately 36.15% are **Adenine (A) or Thymine (T)**.

The sequence therefore has a **relatively high GC content**. GC content is an
important characteristic of DNA sequences and can be used to describe and
compare DNA sequences.

The FASTA parser successfully extracted the sequence identifier, sequence
length, GC content, and the first 50 nucleotides.